src/interfaces/vault.ts

```diff
@@ -46,6 +47,7 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
    */
   async get(addresses?: Address[], overrides?: CallOverrides): Promise<Vault[]> {
     const cached = await this.cachedFetcherGet.fetch();
+    // if (cached && cached.length == -1) {
```

Comment:
@jainkunal I was thinking you could just give this a read, to see if everything makes sense, but if you'd like to test it you need to bypass the cached here and in token.ts.